cdap-common/src/main/java/co/cask/cdap/common/namespace/AbstractNamespaceClient.java

```diff
@@ -37,14 +37,14 @@
   private static final Gson GSON = new Gson();
 
   @Override
-  public void delete(Id.Namespace namespaceId) throws Exception {
-    URL url = resolve(String.format("unrecoverable/namespaces/%s", namespaceId.getId()));
+  public void delete(NamespaceId namespaceId) throws Exception {
```

Comment:
We deprecated Id, but didn't provide cdap-client apis to use EntityId. So its a little tricky